components/skus/browser/rs/lib/src/storage/kv.rs

```diff
@@ -49,38 +55,40 @@ pub trait KVClient {
         Self::Store: KVStore;
 }
 
+#[async_trait(?Send)]
 pub trait KVStore: Sized {
     fn env(&self) -> &Environment;
-    fn purge(&mut self) -> Result<(), InternalError>;
-    fn set(&mut self, key: &str, value: &str) -> Result<(), InternalError>;
-    fn get(&mut self, key: &str) -> Result<Option<String>, InternalError>;
+    async fn purge(&mut self) -> Result<(), InternalError>;
+    async fn set(&mut self, key: &str, value: &str) -> Result<(), InternalError>;
+    async fn get(&mut self, key: &str) -> Result<Option<String>, InternalError>;
 }
 
 fn key_from_environment(env: &Environment) -> String {
     format!("skus:{}", env)
 }
 
+#[async_trait(?Send)]
 impl<C> KVStoreHelpers<C> for C
 where
     C: KVStore,
 {
-    fn get_state(&mut self) -> Result<KVState, InternalError> {
+    async fn get_state(&mut self) -> Result<KVState, InternalError> {
         let key = key_from_environment(self.env());
-        if let Ok(Some(state)) = self.get("rewards:local") {
+        if let Ok(Some(state)) = self.get("rewards:local").await {
             // Perform a one time migration, clearing any old values
-            self.purge()?;
+            self.purge().await?;
             // and setting a new key with the prior value
-            self.set(&key, &state)?;
+            self.set(&key, &state).await?;
         }
-        let state = self.get(&key)?.unwrap_or_else(|| "{}".to_string());
+        let state = self.get(&key).await?.unwrap_or_else(|| "{}".to_string());
         Ok(serde_json::from_str(&state)?)
     }
 
-    fn set_state(&mut self, state: &KVState) -> Result<(), InternalError> {
+    async fn set_state(&mut self, state: &KVState) -> Result<(), InternalError> {
         let key = key_from_environment(self.env());
         event!(Level::DEBUG, "set state");
         event!(Level::TRACE, state = %format!("{:#?}", state), "set state",);
```

Comment:
@evq @nvonpentz do we need to remove that? As that slows down a lot per out tests. Per @evq 
`@serg this is really dumb but can you delete this line and re-run the fetch / grab new logs? https://github.com/brave/brave-core/blob/master/components/skus/browser/rs/lib/src/storage/kv.rs#L82`